src/Symfony/Bundle/FrameworkBundle/Debug/TraceableEventDispatcher.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Debug\CliRequest;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher as BaseTraceableEventDispatcher;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * @internal
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class TraceableEventDispatcher extends BaseTraceableEventDispatcher
+{
+    public function __construct(
+        EventDispatcherInterface $dispatcher,
+        Stopwatch $stopwatch,
+        LoggerInterface $logger = null,
+        private readonly ?RequestStack $requestStack = null
+    ) {
+        parent::__construct($dispatcher, $stopwatch, $logger, $requestStack);
+    }
+
+    /**
+     * Starts a stopwatch section before a command runs to profile listeners.
+     */
+    protected function beforeDispatch(string $eventName, object $event): void
+    {
+        if (!class_exists(ConsoleEvents::class) || !$event instanceof ConsoleCommandEvent) {
```

Comment:
```suggestion
        if (!$event instanceof ConsoleCommandEvent || !class_exists(ConsoleEvents::class)) {
```

Reply:
Shouldn't we use `class_exists` before `instanceof`?

Reply:
why should we? my rationale : instanceof is fast, class_exists not as much

Reply:
Ok, I thought `instanceof` would break if the class did not exist. We don't need the `class_exists` at all then.